server/jetstream_api.go

```diff
@@ -1468,8 +1468,15 @@ func (s *Server) jsStreamUpdateRequest(sub *subscription, c *client, _ *Account,
 		return
 	}
 
+	// Handle clustered version here.
 	if s.JetStreamIsClustered() {
-		s.jsClusteredStreamUpdateRequest(ci, acc, subject, reply, rmsg, &cfg)
+		// If we are inline with client, we still may need to do a callout for stream info
+		// during this call, so place in Go routine to not block client.
+		if c.kind != ROUTER && c.kind != GATEWAY {
```

Comment:
The comment says "if we are inline with client", which I think you mean that the request comes from a CLIENT connection. If so, why not check on `if c.kind == CLIENT`? I mean a "lot" of types are not a CLIENT. We also have ACCOUNT, and LEAF, etc.. so would rather be explicit on what type of CLIENT should do what.

Reply:
For networked accounts I want to make sure we do not block client or leaf. We already do the Go routine dispatch for Routes and Gateways, so wanted to be consistent.